### PR TITLE
Move affinity updating into create_incoming, test Facebook affinity

### PR DESCRIPTION
**Pull request by nicpottier**

### CODE REVIEW

File: temba/contacts/models.py [hunk 1]
```diff
@@ -736,11 +731,6 @@ def get_or_create(cls, org, user, name=None, urns=None, incoming_channel=None, u
                         existing_orphan_urns[urn] = existing_urn
                         if not contact and existing_urn.contact:
                             contact = existing_urn.contact
-
-                    # update this URN's channel
-                    if incoming_channel and existing_urn.channel != incoming_channel:
-                        existing_urn.channel = incoming_channel
-                        existing_urn.save(update_fields=['channel'])
```
Comment (nicpottier) [inline temba/contacts/models.py]: This is now done explicitly in via `ContactURN.update_affinity()` in `Msg.create_incoming`

File: temba/msgs/models.py [hunk 2]
```diff
@@ -1123,10 +1123,13 @@ def create_incoming(cls, channel, urn, text, user=None, date=None, org=None, con
...
         if not contact:
-            contact = Contact.get_or_create(org, user, name=None, urns=[urn], incoming_channel=channel)
+            contact = Contact.get_or_create(org, user, name=None, urns=[urn], channel=channel)
             contact_urn = contact.urn_objects[urn]
         else:
-            contact_urn = None
+            contact_urn = ContactURN.get_or_create(org, contact, urn, channel=channel)
+
+        # check our URN's affinity
+        contact_urn.update_affinity(channel)
```
Comment (nicpottier) [inline temba/msgs/models.py]: It was previously possible to create a Msg object with no URN attached to it. We will now always create a URN to map to the message. This seems right to me but I'm slightly apprehensive about this change as well. Thus far I don't see any test cases that need that behavior though so it certainly feels more right.

File: temba/contacts/models.py [hunk 3]
```diff
@@ -638,7 +638,7 @@ def from_urn(cls, org, urn_as_string, country=None):
             return None
 
     @classmethod
-    def get_or_create(cls, org, user, name=None, urns=None, incoming_channel=None, uuid=None, language=None, is_test=False, force_urn_update=False):
+    def get_or_create(cls, org, user, name=None, urns=None, channel=None, uuid=None, language=None, is_test=False, force_urn_update=False):
```
Comment (nicpottier) [inline temba/contacts/models.py]: No more `incoming_channel`, just `channel`. And no more automagic updating of affinity in `Contact.get_or_create`.

**Comment by nicpottier** [comment]
plz review once tests pass, would like to get this out ASAP

**Comment by norkans7** [comment]
Looks good

**Pull request merged**